sklearn/tests/test_pipeline.py

```diff
@@ -393,6 +394,38 @@ def test_pipeline_methods_preprocessing_svm():
         pipe.score(X, y)
 
 
+def test_pipeline_methods_pca_tsne():
+    # test that a only fit_transform needs to be present in order to
+    # run a pipeline with fit_transform.
+    # Don't require transform to be present, explicitly.
+    pca = PCA(n_components=2, random_state=0)
+    tsne = TSNE(random_state=0)
+    separate_emb = tsne.fit_transform(pca.fit_transform(iris.data))
+
+    pca_for_pipeline = PCA(n_components=2, random_state=0)
+    tsne_for_pipeline = TSNE(random_state=0)
+    msg = ("Intermediate step '%s' (type %s) does not have "
+           "transform, pipeline is not reusable on test data."
+           % (tsne_for_pipeline, type(tsne_for_pipeline)))
+
+    with pytest.warns(UserWarning, match=re.escape(msg)):
+        pipe = make_pipeline(pca_for_pipeline, tsne_for_pipeline,
+                             'passthrough')
+
+    pipeline_emb = pipe.fit_transform(iris.data)
```

Comment:
We should also test that fit_predict works if the final estimator is a clusterer